api/tests/unit/domain/services/obfuscation-service_test.js

```diff
@@ -31,14 +37,14 @@ describe('Unit | Service | user-authentication-method-obfuscation-service', () =
 
   describe('#getUserAuthenticationMethodWithObfuscation', () => {
 
-    it('should return authenticated with samlId when user is authenticated with samlId only',  () => {
+    it('should return authenticated with samlId when user is authenticated with samlId only', async () => {
       // Given
-      const samlId = '1234567';
-      const user = new User({ samlId });
-      user.samlId = '1234567';
+      const user = domainBuilder.buildUser();
+      authenticationMethodRepository.findOneByUserIdAndIdentityProvider.resolves(domainBuilder.buildAuthenticationMethod());
```

Comment:
Ce test et les deux suivants ne passent que parce que le builder d'authenticationMethod créé par défaut une méthode d'authentification GAR.
Du coup, peu importe le userId renseigné, le test sera vert.

Pour éviter des douleurs à ce niveau par la suite, je propose :
```suggestion
      authenticationMethodRepository.findOneByUserIdAndIdentityProvider.resolves(domainBuilder.buildGarAuthenticationMethod({ userId: user.id }));
```

Avec un nouveau builder pour le GAR (puisque celui pour le password existe déjà)

Reply:
Plutôt que de rajouter un builder j'ai préféré renseigner le userId et l'identityProvider GAR.